src/components/DetailsPanelQuery.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+  <div>
+    <v-details-panel-header object-type="SQL query" :title="title" />
+    <v-details-panel-list title="Events" :items="object.events" />
+  </div>
+</template>
+
+<script>
+import VDetailsPanelHeader from '@/components/DetailsPanelHeader.vue';
+import VDetailsPanelList from '@/components/DetailsPanelList.vue';
+import { getSqlLabelFromString } from '@/lib/util';
+import { SELECT_OBJECT } from '../store/vsCode';
+
+export default {
+  name: 'v-details-panel-function',
+  components: {
+    VDetailsPanelList,
+    VDetailsPanelHeader,
+  },
+  props: {
+    object: {
+      type: Object,
+      required: true,
+    },
+  },
+  computed: {
+    title() {
+      return getSqlLabelFromString(this.object.name);
+    },
+
+    events() {
+      return this.$store.state.appMap.events.filter(
+        (e) => e.isCall() && e.codeObject && e.codeObject.id === this.object.id
+      );
+    },
+  },
+
+  mounted() {
+    if (this.object.events.length === 1) {
+      this.$store.commit(SELECT_OBJECT, this.object.events[0]);
```

Comment:
This has the side-effect of leaving the query code object in the selected objects stack:
![image](https://user-images.githubusercontent.com/8737782/112164324-2c6b1c00-8bc4-11eb-9c90-cdc0fa72eda4.png)